Mathlib/Order/Minimal.lean

```diff
@@ -1,472 +1,661 @@
 /-
 Copyright (c) 2022 Yaël Dillies. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
-Authors: Yaël Dillies
+Authors: Yaël Dillies, Peter Nelson
 -/
 import Mathlib.Order.Antichain
 import Mathlib.Order.UpperLower.Basic
 import Mathlib.Order.Interval.Set.Basic
-import Mathlib.Order.RelIso.Set
 
 /-!
-# Minimal/maximal elements of a set
+# Minimality and Maximality
 
-This file defines minimal and maximal of a set with respect to an arbitrary relation.
+This file defines minimality and maximality of an element with respect to a predicate `P` on
+an ordered type `α`.
 
 ## Main declarations
 
-* `maximals r s`: Maximal elements of `s` with respect to `r`.
-* `minimals r s`: Minimal elements of `s` with respect to `r`.
+* `Minimal P x`: `x` is minimal satisfying `P`.
+* `Maximal P x`: `x` is maximal satisfying `P`.
+
+## Implementation Details
+
+This file underwent a refactor from a version where minimality and maximality were defined using
+sets rather than predicates, and with an unbundled order relation rather than a `LE` instance.
```

Comment:
This is the main change that worries me here (all the others look great), in particular this forbids us from (conveniently) talking about minimality with respect to a relation that isn't the default ordering on the type... For the moment, perhaps it's enough to add a TODO saying that this case might need further consideration?

Reply:
Thanks - I've added this TODO to the docstring.